implementations/rust/ockam/ockam_transport_tcp/src/workers/sender.rs

```diff
@@ -212,3 +202,27 @@ impl Worker for TcpSendWorker {
         Ok(())
     }
 }
+
+/// Helper that creates a length-prefixed buffer containing the given
+/// `TransportMessage`'s payload
+///
+/// The length-prefix is encoded as a big-endian 16-bit unsigned
+/// integer.
+fn prepare_message(msg: TransportMessage) -> Result<Vec<u8>> {
+    let mut msg_buf = msg.encode().map_err(|_| TransportError::SendBadMessage)?;
+
+    // Create a buffer that includes the message length in big endian
+    let mut len = (msg_buf.len() as u16).to_be_bytes().to_vec();
+
+    // Fun fact: reversing a vector in place, appending the length,
```

Comment:
This is honestly just cuz of our encoding trait being bad. I indend to rework this eventually, and started on this at one point pre-GA

```rs
// Also, this should be renamed to `Encode` for better rust style...
pub trait Encodable {
    fn encode_into(&self, buf: &mut Encoded) -> Result<()>;
    fn encode(&self) -> Result<Encoded> {
        let mut buf = vec![];
        self.encode_into(&mut buf)?;
        Ok(buf)
    }
}

// And the blanket impl would be something like this
impl<T: Serialize> Encodable for T {
    fn encode_into(&self, buf: &mut Encoded) -> Result<()> {
        use serde_bare::ser::{Serializer, VecWrite};
        let mut serializer = Serializer::new(VecWrite::new(buf));
        self.serialize(&mut serializer)?;
        Ok(())
    }
}
```

But I haven't had time, and we'll be changing encoding formats eventually anyway. It also makes less sense with the `Encoded` type alias (which IMO probably should be removed anyway).